locator/topology.hh

```diff
@@ -71,10 +71,10 @@ public:
     }
 
     const endpoint_dc_rack& get_location(const inet_address& ep) const;
-    sstring get_rack() const;
-    sstring get_rack(inet_address ep) const;
-    sstring get_datacenter() const;
-    sstring get_datacenter(inet_address ep) const;
+    const sstring& get_rack() const;
+    const sstring& get_rack(inet_address ep) const;
+    const sstring& get_datacenter() const;
+    const sstring& get_datacenter(inet_address ep) const;
 
```

Comment:
Even better would be to convert them to pointers to rack/dc objects in token_metadata (then compare pointers), but of course a much bigger change.

Reply:
And should be based on https://github.com/scylladb/scylladb/pull/11987 where node objects would be stable in the topology instance (and stable forever on shard 0)